test/spec/smartbanner_spec.js

```diff
@@ -1,672 +1,984 @@
+      });
+
+      it('expected to work against Android platform', function() {
+        expect(platformOS).to.eql('android');
+      });
+
+      it('expected to have Android price suffix', function() {
+        expect(smartbanner.priceSuffix).to.eql(' - In Google Play');
+      });
+
+      it('expected to return Android template', function() {
+        expect(smartbanner.html).to.eql(ANDROID_BODY_GLOBAL);
+      });
+
+      it('expected to have Android icon', function() {
+        expect(smartbanner.icon).to.eql('icon--google.jpg');
+      });
+
+      it('expected to have Android button URL', function() {
+        expect(smartbanner.buttonUrl).to.eql('https://play.google.com/store');
+      });
+
+    });
+
+  });
+
+  describe('exit', function() {
+
+    context('without jQuery Mobile', function(done) {
+
+      beforeEach(function(done) {
+        jsdom.env({
+          html: HTML,
+          userAgent: USER_AGENT_IPHONE_IOS9,
+          done: function(err, window) {
+            global.document = window.document;
+            global.window = window;
+            global.getComputedStyle = window.getComputedStyle;
+            smartbanner = new SmartBanner();
+            smartbanner.publish();
+            done();
+          }
+        });
+      });
+
+      it('expected to set cookie', function(done) {
+        smartbanner.exit();
+        expect(Bakery.baked).to.be.true;
+        done();
+      });
+
+      it('expected to remove component markup', function(done) {
+        smartbanner.exit();
+        let element = document.querySelector('.js_smartbanner');
+        expect(element).not.to.exist;
+        done();
+      });
+
+      it('expected to restore HTML margin', function(done) {
+        smartbanner.exit();
+        let html = document.querySelector('html');
+        let margin = parseFloat(getComputedStyle(html).marginTop);
+        if (isNaN(margin)) {
+          margin = 0;
+        }
+        expect(margin).to.eql(smartbanner.originalTopMargin);
+        done();
+      });
+    });
+
+    context('with jQuery Mobile', function(done) {
+      before(function(done) {
+        jsdom.env({
+          html: HTML,
+          scripts: SCRIPTS,
+          userAgent: USER_AGENT_IPHONE_IOS9,
+          done: function(err, window) {
+            global.document = window.document;
+            global.window = window;
+            global.$ = window.jQuery;
+            global.getComputedStyle = window.getComputedStyle;
+            smartbanner = new SmartBanner();
+            smartbanner.publish();
+            done();
+          }
+        });
+      });
+
+      it('expected to restore top distance', function(done) {
+        smartbanner.exit();
+        let page = document.querySelector('.ui-page');
+        let top = parseFloat(getComputedStyle(page).top);
+        if (isNaN(top)) {
+          top = 0;
+        }
+        expect(top).to.eql(smartbanner.originalTop);
+        done();
+      });
+    });
+
+  });
+
+  descri
```

Comment:
Open the file to see the full amount of changes, which also includes CRLF to LF change (which is why you might see that all is changed)